src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public boolean add(T value) {
-        return false;
+        return size == 0 ? addFirstValue(value) : addNotFirstValue(value);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Index passed to the method is invalid!");
+        }
+        if (size == index) {
+            add(value);
+        } else {
+            insertByIndex(value, node(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return node(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        T oldValue = node(index).item;
+        Node<T> node = node(index);
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> oldElement = node(index);
+        if (size != 1) {
+            Node<T> beforeToElement = oldElement.prev;
+            Node<T> afterToElement = oldElement.next;
```

Comment:
do we need this variable?

Reply:
I think yes. Because it's easier to understand. For excample "beforeToElement.next = afterToElement" esier to read than "oldElement.prev.next = oldElement.next".

Reply:
ok